Create index.eng.md (#42)
planetarium/snack.planetarium.dev: content/posts/2019/08/pycon-sprint-2019/index.eng.md

---

> title: Libplanet at Pycon 2019 Sprint

Reviewer: They used to capitalize the letter C in the middle.

```suggestion
title: Libplanet at PyCon 2019 Sprint
```

date: 2019-08-21
authors: [dogeon.lee]
translators: [kidon.seo]
---

Hello, I'm Lee Dogeon, [Libplanet] committer at Planetarium!

> This past August 15th and 16th, our team participated at [Pycon 2019 Sprint] as leader of Project [Libplanet]. In this article, I'd like to briefly talk about the contributions we received at Pycon Sprint.

Reviewer: ```suggestion
This past August 15th and 16th, our team participated at [PyCon 2019 Sprint] as leader of Project [Libplanet]. In this article, I'd like to briefly talk about the contributions we received at PyCon Sprint.
```

> [Pycon 2019 Sprint]: https://www.pycon.kr/program/sprint?&lang=en-US

Reviewer: ```suggestion
[Pycon 2019 Sprint]: https://www.pycon.kr/program/sprint?lang=en-US
```

> ## Contributions at Pycon 2019 Sprint

Reviewer: ```suggestion
## Contributions at PyCon 2019 Sprint
```

Thanks to all our participants at the Sprint, many features were added to [Libplanet], as well as [Libplanet Explorer] and [Libplanet Explorer Frontend]!

[Libplanet Explorer] is a project that helps you search blocks created in games developed with Libplanet through [GraphQL], and [Libplanet Explorer Frontend] is a TypeScript + React project to make Libplanet Explorer easier to see in your browser.

![balloon](./balloon.jpg)

> The Pycon Preparation Committee awarded us a balloon whenever we send a PR (pull request) and so I'd like to share some of the PR's that helped us receive a bunch of these balloons at the event!

Reviewer: “Organizing Team” seems their choice of name.  See also: <https://www.pycon.kr/contribute/staff?lang=en-US>.

```suggestion
The PyCon Organizing Team awarded us a balloon whenever we send a <abbr title="pull request">PR</abbr> and so I'd like to share some of the PR's that helped us receive a bunch of these balloons at the event!
```

[GraphQL]: https://graphql.org/

### Suspicious Activity in the Number of Blocks Downloaded

When a node (A) receives a query from another node (B) about a block’s content that the node (B) doesn't have, it (A) divides the block into a certain number and passes the block hash to the node (B) that sent the query.

> But there was a problem in this process in which apart from the first response, the number of block hash received was one less than the expected amount. Contributor [Baek Daehyun][gurrpi] set off on a quest to solve this problem. 

Reviewer: He seems to prefer this spelling for his name:

```suggestion
But there was a problem in this process in which apart from the first response, the number of block hash received was one less than the expected amount. Contributor [Daehyun Paik][gurrpi] set off on a quest to solve this problem. 
```

And thanks to his contribution, we now can send and receive block hashes in the right amount.

[gurrpi]: https://github.com/gurrpi

### Hiding Empty Blocks!

[Libplanet] only accepts the longest chain as [Proof of Work][PoW], so it constantly creates blocks even if there aren’t any transaction. But when searching for transactions rather than the block itself, the query results are mixed with empty blocks, making the process quite cumbersome.

Since the blocks we're interested in are mainly those with transactions, contributor [Ha Hyemi][hyeguiee] added another option to Libplanet through her contribution!

Now we have an option to only see the blocks with transactions.

[PoW]: https://en.bitcoin.it/wiki/Proof_of_work
[hyeguiee]: https://github.com/hyeguiee

### Frontend Docking!

[Docker] is a tool that make it easy to create, deploy, and run applications using containers.

> Contributor [Ryang Minho][minhoryang] helped us create a docker file for [Libplanet Explorer Frontend], providing us with an environment in which we can easily distribute Libplanet Explorer anywhere or run it locally.

Reviewer: He seems to prefer this order for his name:

```suggestion
Contributor [Minho Ryang][minhoryang] helped us create a docker file for [Libplanet Explorer Frontend], providing us with an environment in which we can easily distribute Libplanet Explorer anywhere or run it locally.
```
